be-http.c

```diff
@@ -118,8 +118,8 @@ static int http_post(void *handle, char *uri, const char *clientid, const char *
 		return (FALSE);
 	}
 	if (conf->hostheader != NULL)
-		curl_slist_append(headerlist, conf->hostheader);
-	curl_slist_append(headerlist, "Expect:");
+		headerlist = curl_slist_append(headerlist, conf->hostheader);
+	headerlist = curl_slist_append(headerlist, "Expect:");
```

Comment:
The following would be a further fix, but: aren't there a few return statements in this function before which we should be freeing `headerlist` (and possibly other allocations)?